shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java

```diff
@@ -308,6 +330,67 @@ protected String getDartEntrypointFunctionName() {
   public void onPostResume() {
     Log.d(TAG, "onPostResume()");
     flutterEngine.getLifecycleChannel().appIsResumed();
+
+    // TODO(mattcarroll): find a better way to handle the update of UI overlays than calling through
+    //                    to platformPlugin. We're implicitly entangling the Window, Activity, Fragment,
+    //                    and engine all with this one call.
+    platformPlugin.onPostResume();
```

Comment:
Perhaps we just forward this to flutterEngine, and remove any knowledge of the platformPlugin from this class.

Reply:
We certainly would not forward `onPostResume()` to a `FlutterEngine` because that's an Android lifecycle method.  Even what the `PlatformPlugin` does with this call wouldn't belong in the engine.  I need to think more about this plugin before coming up with an alternative.

Reply:
The engine doesn't necessarily need to know it's an onPostResume, just that something has happened that it needs to inform the PlatformPlugin about.

Reply:
I'll entertain specific suggestions, but anything that comes to my mind would be a violation of encapsulation.  I don't see a legitimate responsibility here for the engine.